### PR TITLE
Roadmap rework

### PR DESCRIPTION
**Pull request by LindsayHill**
I've reworked the roadmap section, since we have not been maintaining it well.

Main points:
* Removed all the "we did this in version <X>" stuff - that all lives in Changelog.
* Reworded a few items, and removed some completed things. Nothing too major. Removed MongoDB item

Things to check when reviewing:
* Are we happy to push 'done' stuff to Changelog?
* Any items we want to add or remove?

Based upon feedback, have removed section headers.

### CODE REVIEW

File: docs/source/roadmap.rst [hunk 1]
```diff
@@ -1,67 +1,52 @@
...
-* **StackStorm Forge:** for Community and Enterprise integration and automation packs. Solve the problem of packs spread all over GitHub. Make integration and automation packs discoverable, continuously tested, and community rated
-* **More integration packs:** push more content to the community to help work with most common and widely used tool. Tell us if there is tool you love and think we should integrate with or better yet write a pack yourself.
-* **Action Output Structure Definition**: enable optional definition of action payload, so that it can be introspected and used when passing data between actions in workflows.
-* **Datastore:**  Hierarchical, pluggable and secure datastore to bring together all the gnarly config and key spread in operations environments.
-* **Documentation:** generate REST API docs.
-* **Flow v2:** Visualizing workflow executions. UI improvements, including more "operator" persona features.
+.. _platform:
 
+Core Platform
+-------------
```
Comment (dzimine) [inline docs/source/roadmap.rst]: I'm -1 on introducing categories: many things will fall behind, and our community content is dynamic and situational. Let's stay with one list.

File: docs/source/roadmap.rst [hunk 2]
```diff
@@ -1,67 +1,52 @@
...
+* **History and Audit service:** History view with advanced search over years worth of execution records, over multiple versions of continuously upgraded StackStorm.
+* **Migrate away from MongoDB:** Replace Mongo, to reduce operational overhead, improve performance, and simplify licensing.
+* **At-scale refinements:** Ensure event handling reliability, and event storm resilience. Complete support for multi-node deployment of sensor containers and rules engines for resilience and throughput.
+* **Security hardening:** Complete security audit and address issues discovered so far.
 
-.. rubric:: Done in v1.1
+.. _content:
 
-* **FLOW:** Visual workflow representation and drag-and-drop workflow designer.
-* **RBAC:** Role based access control for packs, actions, triggers and rules.
```
Comment (dzimine) [inline docs/source/roadmap.rst]: keep RBAC v2 as a big TODO.

File: docs/source/roadmap.rst [hunk 3]
```diff
@@ -1,67 +1,52 @@
...
-.. rubric:: Done in v0.11
+.. _rbac:
 
-* **ChatOps:** two-way chat integration beyond imagination.
-* **More integration packs**: Major integrations - Salt, Ansible, some significant others. `Check the full list <https://github.com/StackStorm/st2contrib/tree/master/packs>`_.
+RBAC
+----
 
-.. rubric:: Done in v0.9
+* **Filters**: Tag and property based filters, more refined and convenient access control
```
Comment (dzimine) [inline docs/source/roadmap.rst]: here is an example for "no rubrics": filters on tags and properties are not in RBAC category but they touch it, as many others.

File: docs/source/roadmap.rst [hunk 4]
```diff
@@ -1,67 +1,52 @@
...
-* **Flow v2:** Visualizing workflow executions. UI improvements, including more "operator" persona features.
+.. _platform:
 
+Core Platform
+-------------
 
-See :doc:`/changelog` for details on what is done.
+* **Docker based installer:** Complete the vision of OS independent, layered Docker-based installer, to increase reliability, modularity, and speed of deployment.
+* **History and Audit service:** History view with advanced search over years worth of execution records, over multiple versions of continuously upgraded StackStorm.
+* **Migrate away from MongoDB:** Replace Mongo, to reduce operational overhead, improve performance, and simplify licensing.
```
Comment (dzimine) [inline docs/source/roadmap.rst]: I am tempted to ~~cross it out~~

Comment (Kami) [inline docs/source/roadmap.rst]: Agreed :P

Maybe also add a note that since v1.6.0 we support MongoDB 3.2.

**Comment by dzimine** [comment]
@LindsayHill thanks for doing this. Putting me to shame as I have it on my TODO list, along with refreshed roadmap items.

OK to push "done" to Changelog, although I admit the granularity of change log doesn't provide the sense of progress I was trying to achieve here... but Ok.

-1 on section headers: suggest we get rid of them all together. 

I'll look up my list in the office for "add or remove". Can do commit on top. This is still better when it is. 

You should cherry-pick this to v1.6 else you (we) will be irritated with outdated roadmap for 2 more weeks :)

**Comment by dzimine** [comment]
Feel free to merge if you remove section headers.

**Comment by armab** [comment]
Closes [#225: Roadmap page is totally Outdated](https://github.com/StackStorm/st2docs/issues/225)

File: docs/source/roadmap.rst (hunk 1 above)
Comment (LindsayHill) [inline docs/source/roadmap.rst]: We fell behind with one list. Not sure categories make that better or worse. But will change to one list.

File: docs/source/roadmap.rst (hunk 4 above)
Comment (LindsayHill) [inline docs/source/roadmap.rst]: Will do

**Comment by LindsayHill** [comment]
Updated PR based upon feedback. Removed section headers (nb have maintained RBAC grouping, that's what we had before). Removed MongoDB item.

File: docs/source/roadmap.rst [hunk 5]
```diff
@@ -1,67 +1,30 @@
...
-* **All-in-one installer**: production ready single-box reference deployment with graphical setup wizard.
-* **RHEL 6 and 7 support**
-* **Trace-tags**: ability to track a complete chain of triggers, rules, executions, related to a given triggering event.
-* **Native SSH:** replace Fabric; Fabric based SSH still available and can be enabled via config.
-* **WebUI major face-lift**
-
-
-.. rubric:: Done in v0.11
-
-* **ChatOps:** two-way chat integration beyond imagination.
```
Comment (armab) [inline docs/source/roadmap.rst]: Linking to st2 [`changelog`](https://docs.stackstorm.com/changelog.html) is great, but I really liked high-level overview of what was done in StackStorm as a _Major_ progress.

I think that st2 users can appreciate it too, especially ones that don't update StackStorm frequently or still considering using StackStorm and generally curious about progress.

^^ That information is pretty much impossible to get from the low-level & detailed [CHANGELOG.rst](https://github.com/StackStorm/st2/blob/master/CHANGELOG.rst)

-----

If we follow "old" approach, then it's additional work to add major changes in recent versions.

Comment (LindsayHill) [inline docs/source/roadmap.rst]: > then it's additional work to list major changes
Yeah. And that's what we've not been good at doing. That's why I'd rather remove it from here. Plus roadmap should be where we're *going*, not where we've been. Perhaps we need some sort of "Release History" doc that summarises major changes. Would not include small changes or bugfixes.

Comment (armab) [inline docs/source/roadmap.rst]: Here are the changes I briefly extracted from the blog posts, thanks to guys for writing Release Announcements: 

* `1.6`: https://stackstorm.com/2016/08/10/stackstorm-v1-6-0/
 * MongoDB 3.x support
 * Access K/V datastore from the Mistral workflows
 * Pack Configuration Validation

* `1.5`: https://stackstorm.com/2016/06/27/stackstorm-v1-5-alive/
  * New Pack Configuration separated from the pack code
  * CLI Automcompletion
  * Datastore Secrets

* `1.4`: https://stackstorm.com/2016/04/20/stackstorm-v1-4/
  * Deprecation of All-in-One Installer
  * Native `deb`/`rpm` packages with bundled python dependencies
  * Action aliases testing
  * ChatOps Slack API support

* `1.3`: https://stackstorm.com/2016/01/29/stackstorm-1-3-0-is-here/
  * `st2 re-run` - resume failed workflows
  * Garbage collection service
  * More debugging options

* `1.2`: https://stackstorm.com/2015/12/08/stackstrom-1-2-release-announcement/
  * Pack Testing
  * Fully reworked ChatOps with Jinja templating
  * Timeout and retry policies

Let's, at least, include that.
@emedvedev @Kami @lakshmi-kannan let me know if I missed some big feature in ^^ that list.

--------

I'm sure that keeping high-level StackStorm history in the `Roadmap` is the right thing to do. Without history aka "what's done", future plans look like empty words.

Comment (Kami) [inline docs/source/roadmap.rst]: Sounds good to me :+1:

Comment (LindsayHill) [inline docs/source/roadmap.rst]: I'll add past updates back into the roadmap. I'm still not convinced that the past belongs in a 'futures' document though. From a cursory search, most other projects that publish a roadmap do not use it to refer to previous updates.

**Comment by armab** [comment]
@LindsayHill Special thanks for keeping the history!
Really hope we do better for the end-user.

**Pull request merged**